var/spack/repos/builtin/packages/zfp/package.py

```diff
@@ -31,23 +31,41 @@ class Zfp(Package):
        target error bounds or bit rates.  Although bit-for-bit lossless
        compression is not always possible, zfp is usually accurate to
        within machine epsilon in near-lossless mode, and is often orders
-       of magnitude more accurate than other lossy compressors.
-
+       of magnitude more accurate than other lossy compressors. Versions
+       of zfp 0.5.1 or newer also support compression of integer data.
     """
 
     homepage = "http://computation.llnl.gov/projects/floating-point-compression"
-    url      = "http://computation.llnl.gov/projects/floating-point-compression/download/zfp-0.5.0.tar.gz"
+    url      = "http://computation.llnl.gov/projects/floating-point-compression/download/zfp-0.5.1.tar.gz"
 
+    version('0.5.1', '0ed7059a9b480635e0dd33745e213d17')
     version('0.5.0', '2ab29a852e65ad85aae38925c5003654')
 
+    variant('bswtuint8', default=False,
+            description='Build with bit stream word type of uint8')
+
+    def edit(self, spec, prefix):
+        config_file = FileFilter('Config')
+
+        if '+bswtunit8' in self.spec:
+            config_file.filter('#DEFS += -DBIT_STREAM_WORD_TYPE=uint8)',
+                               'DEFS += -DBIT_STREAM_WORD_TYPE=uint8')
+
     def install(self, spec, prefix):
         make("shared")
 
+        zfp_incdir = 'include' if spec.satisfies('@0.5.1:') else 'inc'
+
         # No install provided
         mkdirp(prefix.lib)
         mkdirp(prefix.include)
         install('lib/libzfp.so', prefix.lib)
-        install('inc/zfp.h', prefix.include)
-        install('inc/types.h', prefix.include)
-        install('inc/bitstream.h', prefix.include)
-        install('inc/system.h', prefix.include)
+        install('%s/zfp.h' % zfp_incdir, prefix.include)
+        install('%s/bitstream.h' % zfp_incdir, prefix.include)
+        if spec.satisfies('@0.5.1:'):
+            mkdirp('%s/zfp' % prefix.include)
+            install('%s/zfp/system.h' % zfp_incdir, '%s/zfp' % prefix.include)
+            install('%s/zfp/types.h' % zfp_incdir, '%s/zfp' % prefix.include)
+        else:
+            install('%s/types.h' % zfp_incdir, prefix.include)
+            install('%s/system.h' % zfp_incdir, prefix.include)
```

Comment:
If you think there are files in their that could possibly break the installation and you really don't want them in there, then installing them one at a time works. It's just a lot more tedious and changes with every release (as you've noticed).